### DISCUSSION TITLE
Translate Toolbar Buttons Placeholder

### DISCUSSION THREAD

**Discussion by arselane** [sibiraj-s/ngx-editor#566] [Q&A] [answered]
Hi I'm wondering if it's possible to custumise the Toolbar buttons placeholder, 
when I hover on a button a text is shown, for example button "B" => 'Bold', and I'd like to translate it in french wich will be 'Gras"

Thanks in advance

**Comment by michele-cassani** [comment]
Hi if you need a static translation you can follow the documentation: https://sibiraj-s.github.io/ngx-editor/configuration/

If you also need a dynamic translation using the translateService of the @ngx-translate there is also a way but in that case I have to write here because it's not documented

**Reply by sibiraj-s** [reply]
Yes. That is correct.

**Reply by arselane** [reply]
Thanks for your answer. If I get it rigth I'll need to use the placeholder property ?
![image](https://github.com/user-attachments/assets/0943038f-c6f6-47dc-8e14-34c40da858e2)

cause when I'm changing in the locals{} object it does nothing and stays in english
![image](https://github.com/user-attachments/assets/86b9744d-c8f2-40d6-a9cf-9def85512ea4)

thanks again

**Reply by sibiraj-s** [reply]
My bad. I misunderstood the question, yes it should work, let me check once.

**Reply by sibiraj-s** [reply, marked as answer]
<img width="174" alt="Screenshot 2024-07-30 at 7 18 45 PM" src="https://github.com/user-attachments/assets/f46ea0f5-7ed6-47d8-a565-29fbb189d9fe">

Works for me, My simple config

![code](https://github.com/user-attachments/assets/b811fbc7-dfe1-4e1c-a004-a4bc41470619)

**Reply by arselane** [reply]
Thanks a lot, now I know exactly what approach to take

Take care

**Reply by totodej** [reply]
Hi, 
I am interested on this topic.
I would like to know where do you put your appConfig function ?
And where you call this function ?

Thanks in advance

**Reply by sibiraj-s** [reply]
You can refer the demo folder available in this repository. 

https://github.com/sibiraj-s/ngx-editor/blob/master/projects/demo/src/app/app.config.ts

**Reply by mgohin** [reply]
> Hi if you need a static translation you can follow the documentation: https://sibiraj-s.github.io/ngx-editor/configuration/
> 
> If you also need a dynamic translation using the translateService of the @ngx-translate there is also a way but in that case I have to write here because it's not documented

Could you provide the example with `@ngx-translate` please ?

**Comment by sibiraj-s** [comment]
You can use custom icons, https://sibiraj-s.github.io/ngx-editor/menu/#custom-icons-in-ngx-editor-menu, was never intended for text content, but you can have svg with text, will that help?
